nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/framework/configuration/FlowControllerConfiguration.java

```diff
@@ -471,4 +479,38 @@ public AssetSynchronizer assetSynchronizer() throws Exception {
     public AssetComponentManager affectedComponentManager() throws Exception {
         return new StandardAssetComponentManager(flowController());
     }
+
+    /**
+     * Flow Action Reporter configured from NiFi Application Properties
+     *
+     * @return Flow Action Reporter
+     * @throws Exception Thrown on failures to create Flow Action Reporter
+     */
+    @Bean(destroyMethod = "preDestruction")
+    public FlowActionReporter flowActionReporter() throws Exception {
+        final String configuredClassName = properties.getProperty(FLOW_ACTION_REPORTER_IMPLEMENTATION, DEFAULT_FLOW_ACTION_REPORTER_IMPLEMENTATION);
+        if (StringUtils.isBlank(configuredClassName)) {
+            throw new RuntimeException("Cannot create FlowActionReporter because the configuration is missing the following property: "
+                + FLOW_ACTION_REPORTER_IMPLEMENTATION);
+        }
+        try {
+            FlowActionReporter reporter = NarThreadContextClassLoader.createInstance(extensionManager, configuredClassName, FlowActionReporter.class, properties);
+            reporter.onConfigured(new StandardFlowActionReporterConfigurationContext(sslContext, trustManager));
+            return reporter;
+        } catch (final Exception e) {
+            throw new RuntimeException("Failed to create FlowActionReporter with class " + configuredClassName, e);
+        }
+
+    }
+
+    /**
+     * Action Converter for converting actions to flow actions
+     *
+     * @return Action Converter
+     * @throws Exception Thrown on failures to create Action Converter
+     */
+    @Bean
+    public ActionConverter flowActionConverter() throws Exception {
```

Comment:
It seems unnecessary to define this as a Bean, I recommend instantiating the implementation directly in the `auditService()` method.